lib/semian/redis.rb

```diff
@@ -111,6 +111,10 @@ def raise_if_out_of_memory(reply)
       return unless reply.message =~ /OOM command not allowed when used memory > 'maxmemory'\.\s*\z/
       raise ::Redis::OutOfMemoryError.new(reply.message)
     end
+
+    def dns_resolve_failure?(e)
+      e.to_s.match?(/(can't resolve)|(name or service not known)|(nodename nor servname provided, or not known)|(failure in name resolution)/i)
```

Comment:
Note, #match? is a 2.4 feature, but 2.4 is from 2016 and 2.3 is end-of-life now.